### DISCUSSION TITLE
Any initiative to implement columns in Marp?

### DISCUSSION THREAD

**Discussion by cunhapaulo** [marp-team/marp#192] [Ideas] [answered]
Columns are a integral and frequent feature in slide creation. Is there a way to make slides in Marp with two or more columns?

Thanks in advance!

**Comment by yhatt** [comment, marked as answer]
https://twitter.com/y_hatt/status/1449951469961023488

## CSS grid

```markdown
---
marp: true
style: |
  .columns {
    display: grid;
    grid-template-columns: repeat(2, minmax(0, 1fr));
    gap: 1rem;
  }
---

# Multi columns in Marp slide

<div class="columns">
<div>

## Column 1

Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptas eveniet, corporis commodi vitae accusamus obcaecati dolor corrupti eaque id numquam officia velit sapiente incidunt dolores provident laboriosam praesentium nobis culpa.

</div>
<div>

## Column 2

Tempore ad exercitationem necessitatibus nulla, optio distinctio illo non similique? Laborum dolor odio, ipsam incidunt corrupti quia nemo quo exercitationem adipisci quidem nesciunt deserunt repellendus inventore deleniti reprehenderit at earum.

</div>
</div>
```

<details>
<summary>Use <a href="https://tailwindcss.com">Tailwind CSS v2</a> utilities (obsoleted)</summary>

```markdown
---
marp: true
style: @import url('https://unpkg.com/tailwindcss@^2/dist/utilities.min.css');
---

# Multi columns in Marp slide

<div class="grid grid-cols-2 gap-4">
<div>

## Column 1

Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptas eveniet, corporis commodi vitae accusamus obcaecati dolor corrupti eaque id numquam officia velit sapiente incidunt dolores provident laboriosam praesentium nobis culpa.

</div>
<div>

## Column 2

Tempore ad exercitationem necessitatibus nulla, optio distinctio illo non similique? Laborum dolor odio, ipsam incidunt corrupti quia nemo quo exercitationem adipisci quidem nesciunt deserunt repellendus inventore deleniti reprehenderit at earum.

</div>
</div>
```

> I think using Tailwind CSS is a good choice only in PDF conversion because it has no solutions to apply optimizations for now. [Tailwind CSS does not recommend using CSS through CDN in standard use due to a huge filesize (about 3MB in raw).](https://v2.tailwindcss.com/docs/installation#using-tailwind-via-cdn) If hosted presentation HTML on the Web, the score of page speed will be decrease, so should be better to use an inline style instead.
>
> Of course, it is still a tempting option if you does not mind this fact and like to utility-based CSS rather than the classic inline style.
>
> For developers, `marp(it)-tailwind` may become a potentially plugin idea for Marp, to make deep integration with optimized Tailwind CSS.

</details>

Past discussion is available in https://github.com/marp-team/marpit/issues/137.

**Reply by cunhapaulo** [reply]
Thank you so much for your amazing effort!

**Reply by cunhapaulo** [reply]
I don't know why, but it didn't funtion well in my VS Code visualization panel.
I copied and pasted the code so that I wouldn't mistype anything, but it didn't work.

**Reply by yhatt** [reply]
Marp is safety first. HTML tags cannot render by default. Turn on `Markdown > Marp: Enable HTML` from VS Code settings.

![](https://user-images.githubusercontent.com/3993388/57288647-d76bad00-70f4-11e9-8810-0d8987363f8b.png)

**Reply by cunhapaulo** [reply]
Thank you again! 
Wish I can give back to the project and your effort!

**Reply by sbibauw** [reply]
Thanks, @yhatt, for the solution and for the idea of using TailwindCSS. It never occurred to me, but that's a hugely powerful tool to implement more advanced styling in Marp! I feel the possibilities are endless, but there are some powerful uses that would benefit from examples (e.g., for styling "outline/ToC" kind of slides, cover/header slides...). Have you documented Tailwind's use more thoroughly?

**Reply by samlevan** [reply]
@yhatt just a thank you for sharing this solution, it is almost magical. 🙌

**Reply by Compizfox** [reply]
I just tried this, but it appears that markdown inside such a column is not parsed, only HTML. On further thought, that does makes sense (since it is inside a HTML block), but it's a major drawback in my opinion, as a big part of Marp's appeal is the Markdown syntax...

**Reply by Compizfox** [reply]
Something else that's strange:

When there is no blank line between the `<div>` and the text, it is rendered left-aligned:
```
<div class="columns">
<div>
Tempore ad exercitationem necessitatibus nulla, optio distinctio illo non similique? Laborum dolor odio, ipsam incidunt corrupti quia nemo quo exercitationem adipisci quidem nesciunt deserunt repellendus inventore deleniti reprehenderit at earum.
</div>
<div>
Tempore ad exercitationem necessitatibus nulla, optio distinctio illo non similique? Laborum dolor odio, ipsam incidunt corrupti quia nemo quo exercitationem adipisci quidem nesciunt deserunt repellendus inventore deleniti reprehenderit at earum.
</div>
</div>
```
![image](https://user-images.githubusercontent.com/7603719/184147553-7ef2d779-b055-443d-af92-f26390bc5d68.png)

But when there is, it is rendered centered:

```
<div class="columns">
<div>

Tempore ad exercitationem necessitatibus nulla, optio distinctio illo non similique? Laborum dolor odio, ipsam incidunt corrupti quia nemo quo exercitationem adipisci quidem nesciunt deserunt repellendus inventore deleniti reprehenderit at earum.
</div>
<div>

Tempore ad exercitationem necessitatibus nulla, optio distinctio illo non similique? Laborum dolor odio, ipsam incidunt corrupti quia nemo quo exercitationem adipisci quidem nesciunt deserunt repellendus inventore deleniti reprehenderit at earum.
</div>
</div>
```

![image](https://user-images.githubusercontent.com/7603719/184147475-7354bdb1-4927-4f9b-9bb0-9e383ce78406.png)

**Reply by yhatt** [reply]
> I just tried this, but it appears that markdown inside such a column is not parsed, only HTML.

@Compizfox I guess you are missing blank lines between HTML block and Markdown. You should always add blank lines because _they are important to separate both syntax correctly._

[CommonMark spec](https://spec.commonmark.org/0.30/) says:

> it provides a very simple and flexible way of including Markdown content inside HTML tags: simply separate the Markdown from the HTML using blank lines:
>
> [![commonmark spec](https://user-images.githubusercontent.com/3993388/184155321-69a9e4cc-9f98-4329-b8b1-a8999b9e27d6.png)](https://spec.commonmark.org/0.30/#example-188)
>
> _&mdash; [CommonMark Spec](https://spec.commonmark.org/0.30/#html-blocks:~:text=it%20provides%20a%20very%20simple%20and%20flexible%20way%20of%20including%20Markdown%20content%20inside%20HTML%20tags%3A%20simply%20separate%20the%20Markdown%20from%20the%20HTML%20using%20blank%20lines%3A)_

Actually the second example you have provided seems to be already recognized a paragraph as Markdown by separating HTML block and Markdown with a blank line. Thus, you should be able to use Markdown syntaxes such as `*Emphasized*`.

Paragraphs in Markdown will be wrapped with `<p>` tag. ([CommonMark spec](https://spec.commonmark.org/0.30/#example-219)) So I expect the centered text is providing by the style for `<p>` tags in a theme you are using. Aren't you using a theme or class that will align Markdown text to center?

**Reply by Compizfox** [reply]
@yhatt I see, thanks! In a way my second issue was actually the solution to my first, lol.

**Reply by lpinilla** [reply]
I've been trying this answer but it doesn't work for me.

I'm using marp `v2.2.0` 

What I've done:

```
---
marp: true
style: |
  .columns {
    display: grid;
    grid-template-columns: repeat(2, minmax(0, 1fr));
    gap: 1rem;
  }
---
```

At the top of the md file.

My last slide is:

```
---

# Multi columns in Marp slide

<div class="columns">
<div>

## Column 1

Content 1

</div>
<div>

## Column 2

Content2

</div>
</div>
---
```

This is the result:

![image](https://user-images.githubusercontent.com/10984467/192035298-c5a4f303-c5a4-4ca8-9674-f1673b8aee3f.png)


I'll also take the chance to ask: what if I wanted 3 divisions instead of 2?

**Reply by codingluke** [reply]
@lpinilla are you sure you enabled HTML with `--html` or in vs-code like [@yhatt already stated](https://github.com/orgs/marp-team/discussions/192#discussioncomment-1517399)?

To have 3 columns just change `grid-template-columns: repeat(2, minmax(0, 1fr));` to `grid-template-columns: repeat(3, minmax(0, 1fr));` and add another `</div><div>`

**Reply by lpinilla** [reply]
:man_facepalming:  I didn't add the `--html` flag, I was confused because I wanted to export it to pdf so I didn't use that flag. Thanks for the quick reply!

**Reply by lpinilla** [reply]
and what about giving each column a different background color / outline ? is that possible?

**Reply by yhatt** [reply]
> and what about giving each column a different background color / outline ? is that possible?

It's very easy. Set the class to each inner `<div>`s for specific styles.

```markdown
---
marp: true
style: |
  .columns {
    display: grid;
    grid-template-columns: repeat(2, minmax(0, 1fr));
    gap: 1rem;
  }
  .columns-left {
    background: yellow;
  }
  .columns-right {
    background: beige;
  }
---

<div class="columns">
<div class="columns-left">

## Column 1

Content 1

</div>
<div class="columns-right">

## Column 2

Content2

</div>
</div>
```

<img src="https://user-images.githubusercontent.com/3993388/194137393-0d803240-c27a-4bc5-b86b-c4385e5514ad.png" width="400" />

> Or you also can use [`:nth-child(n)` pseudo-class](https://developer.mozilla.org/ja/docs/Web/CSS/:nth-child) to target the single column without setting another class name.
>
> ```css
> .columns {
>   display: grid;
>   grid-template-columns: repeat(2, minmax(0, 1fr));
>   gap: 1rem;
> }
> .columns > :nth-child(1) {
>   background: yellow;
> }
> .columns > :nth-child(2) {
>   background: beige;
> }
> ```

**Reply by kirnhans** [reply]
This is really cool, thank you for this!

**Comment by codingluke** [comment]
@yhatt I just had an idea about this topic. I understand that it ist not easy compatible with common markdown to implement multi column natively. Then it came to my mind, what if there is a page derictive `multi-slide: number`, **which will render the current slide on the left and the next `n` slides to the right**. So the two columns would be just slides rendered next to each other? this would be compatible with common markdown and also extendable with additional layouts. What do you think about it?

**Reply by yhatt** [reply]
We do never want to force Marp specific syntax for the layout. Marp team prefer using combination of Web standards, HTML + CSS, than adding Marp-specific directive and usage.

**Reply by codingluke** [reply]
I totally get this, and your `columns` class example ist just fine for me.  Thanks! 🙇 

However, what is your opinion about conventions like `![bg right](image-path)`, **which I loved since Deckset**. Isn't this somehow a specific syntax for the layout? 😝

**Reply by yhatt** [reply]
[A former Marp app](https://github.com/yhatt/marp) had designed as an open-source and cross-platform version of Deckset, so `left` and `right` keyword for the image syntax had been implemented to Marp as a compatible syntax/feature with Deckset as a result. Actually these keywords are proven effective for making a slide by Deckset.

Similarly, Marp requires to confirm proved effectivity of the new syntax, through third-party Marp/Marpit plugin and several use cases in a real world, for adding syntax as a built-in feature. [See also our contributing guideline.](https://github.com/marp-team/.github/blob/master/CONTRIBUTING.md#before-adding-new-feature)

**Comment by codingluke** [comment]
I cust did a little "hack" with the proposed columns class and the [`markdown-it-container`](https://github.com/markdown-it/markdown-it-container) plugin.

```javascript
// engine.js
const markdownItContainer = require('markdown-it-container');

module.exports = ({ marp }) => marp
  .use(markdownItContainer, 'columns', {
    render: function (tokens, idx) {
      if (tokens[idx].nesting === 1) {
        return '<div class="columns"><div>\n';
      } else {
        return '</div></div>\n';
      }
    }
  })
  .use(markdownItContainer, 'split', {
    render: function (tokens, idx) {
      if (tokens[idx].nesting === 1) {
        return '</div><div>\n';
      } else {
        return '</div></div>\n';
      }
    }
  })
```

With this engine it is possible to write columns like this.

```markdown

# Heading

:::columns

left content

:::split

right content

:::
```

it will be rendered to

```html
<h1>Heading</h1>
<div class="columns"><div>

left content

</div><div>

right content

</div></div>
```

**Comment by kazumatu981** [comment]
FYI: I've been build short cut as plugin!
please enjoy!

[@kazumatu981/markdown-it-nested-container](https://www.npmjs.com/package/@kazumatu981/markdown-it-nested-container)

**Comment by roland-KA** [comment]
I would like to see a solution as suggested in https://github.com/marp-team/marpit/issues/137 using just a *** separator in Markdown and no additional HTML directives.

One reason for this is the following: I'm using Typora for editing Markdown and apart from creating slides with marp I'm using also Typoras export feature for creating PDFs (in A4 format). I've noticed that when I use HTML `<div ..>`in my Markdown documents (for creating columns), the PDF export of Typora gets distorted (the different heading levels get wrong formatting).

**Reply by yhatt** [reply]
So how about porting behavior Deckset's `[.column]` to `***`?
https://docs.deckset.com/English.lproj/Formatting/14-columns.html

<table>
<tr>
<th>Deckset</th>
<th>Suggestion<br>(CommonMark compliant)</th>
</tr>
<tr>
<td>

```markdown
# First page

[.column]

## The first column

Lorem ipsum

[.column]

## The second column

Lorem ipsum

---

# Second page
```

</td>
<td>

```markdown
# First page

***

## The first column

Lorem ipsum

***

## The second column

Lorem ipsum

---

# Second page
```

</td>
</table>

The first `***` in a page indicates beginning of columns. After the first `***`, all rest contents in the page will be trapped in the column, and the author can make more columns by extra splitters `***`. Each columns have the same width by divided evenly in the available space.

This syntax cannot define ending of columns. The solution of using HTML with CSS grid is still available for making highly customized columns. (e.g. append contents after the columns, set custom width for each columns, set background colors, etc...)

**Reply by drbeco** [reply]
I say go for it.... It is a great solution, simple and easy, to a problem that needs addressing.

**Reply by drbeco** [reply]
Is there a way to end the slide in a single column again, after bifurcating? Sometimes it is so useful to write a sentence or a note that is independent of the columns.

Example (using tables, but consider columns):

---
## Header single column

text single column

col1 | col2 | col3
---|---|---
data | date | dati |
dota | dute | deto |

text single column

---

**Reply by sbibauw** [reply]
If the purpose is to add a note, I'd simply use the `_footer` option for this specific slide. That should be the easiest option.

`<!-- _footer: "Note: Those are all the data we have." -->`

**Reply by drbeco** [reply]
Not really. I edited to avoid confusion. Thanks for the feedback, I know how to add notes and it is a cool feature, but unrelated here.

**Reply by yhatt** [reply]
I have no idea. Even if defined a columns ending syntax from CommonMark compliant syntax, I think that would become not intuitive as Markdown/Marp document.

[Using HTML within Markdown is CommonMark compliant way.](https://spec.commonmark.org/0.29/#html-blocks) [As I said earlier](https://github.com/orgs/marp-team/discussions/192#discussioncomment-6395119:~:text=This%20syntax%20cannot%20define%20ending%20of%20columns.%20The%20solution%20of%20using%20HTML%20with%20CSS%20grid%20is%20still%20available%20for%20making%20highly%20customized%20columns), `<div class="columns">` should be working for that advanced case.

```markdown
---
marp: true
style: |
  .columns {
    display: grid;
    grid-template-columns: repeat(3, minmax(0, 1fr));
    gap: 1rem;
  }
---

# Header single column

text single column

<div class="columns">
<div>

## Column 1

</div>
<div>

## Column 2

</div>
<div>

## Column 3

</div>
</div>

text single column

---

...
```

**Comment by thomaseizinger** [comment]
I've found the following to work really well for me:

```html
<style scoped>
section {
    columns: 2;
}

h1 {
    column-span: all;
}

h2 {
    break-before: column;
}
</style>
```

Paste this at the top of a slide you want to have columns in. This assumes that each columns has an `h2` heading and all content after it will be in that column. On the next h2 element, a new column will be created. You will need to also specify the number of columns you want to have.

**Reply by vfrolov** [reply]
This looks like it should work but it doesn’t.

```
---
<style scoped>
section {
    columns: 2;
}
h1 {
    column-span: all;
}
h2 {
    break-before: column;
}
</style>

# Slide Title

## Column 1

This goes in the left column

## Column 2

This goes in the right column
```

Anything wrong with the above?

**Reply by thomaseizinger** [reply]
You may need a block element as column content. I only tested it with a list, i.e. `<ul>`.

**Reply by vfrolov** [reply]
Nope, this doesn’t seem to work with any elements.

**Reply by yhatt** [reply]
The `columns` property requires the applied container element to be [a block layout](https://developer.mozilla.org/en-US/docs/Glossary/Block-level_content). If the applied `section` uses Flexbox layout or Grid layout, the `columns` property will not work.

The default theme of Marp applies `display: flex` to the `section` element in order to achieve vertical center alignment of contents using Flexbox. Therefore, to make the `columns` property work correctly, you should either apply `display: block` to the `section` to disable Flexbox or use a theme that doesn't use Flexbox, such as the `gaia` theme.

```diff
 <style scoped>
 section {
   columns: 2;
+  display: block;
 }
 
 h1 {
   column-span: all;
 }
 
 h2 {
   break-before: column;
 }
 </style>
```

> [!TIP]
>
> [Baseline 2024](https://web.dev/baseline/2024) browsers allow [centering contents vertically within the block layout](https://web.dev/blog/align-content-block). In new [Marp Core v4](https://github.com/marp-team/marp-core/releases/tag/v4.0.0), the default style of slide containers will be `display: block`, that is compatible with `columns` CSS property + vertical centering behavior.
>
> - #533

**Reply by thomaseizinger** [reply]
Ah yes, I am using `gaia` as my theme :)

**Reply by drbeco** [reply]
This is such a nice solution!

**Comment by DylanRJohnston-FZ** [comment]
I ended up just using CSS Grid
```
  section {
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-content: center;
  }

  marp-pre {
    margin: 0px
  }
```

**Comment by codingluke** [comment]
The problem with the grid on the `section` element is, that every html child tag of it is a grid item. This make it possible to do two column, but only with two elements.

For me also the [flex solution](https://github.com/orgs/marp-team/discussions/192#discussioncomment-6023390) is not working always as expected with images on the right side. 

For the meantime the original solution with the additional `<div class="columns">` is maybe polluting the markdown but is at the end the best predictable.

**Reply by thomaseizinger** [reply]
Yes. What is nice about the CSS `columns` is that you can declare your content top to bottom and each new `h2` is a new column.

But, I also just realized this isn't supported in Firefox & Safari yet so you'll have to use a Chromium based browser for viewing the presentation.

**Reply by codingluke** [reply]
and to fix Typora what about [my solution](https://github.com/orgs/marp-team/discussions/192#discussioncomment-3543087) here? is it somehow possible to disable some elements starting with ::: in the pdf creation? There is also [nota.md](https://nota.md/), which does not reformat and destory html in md files.

**Comment by JIMMYGENN** [comment]
I have known how to split two and three columns by `<div class="columns">`. I still wanna know how to split multiple columns with different widths. The default setting of the width of two columns is 1:1, but I want to set it to 7:3. How can I do that? @yhatt

**Reply by yhatt** [reply]
```css
display: grid;
grid-template-columns: minmax(0, 7fr) minmax(0, 3fr);
```

Learn about `fr` unit in CSS grid: https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Grids#flexible_grids_with_the_fr_unit

**Reply by JIMMYGENN** [reply]
It works! Thanks @yhatt.

**Comment by phifuh** [comment]
@yhatt I totaly disagree with your stance on [other issue](https://github.com/marp-team/marpit/issues/137) that this feature is pure styling and everyone needs to figure out a style via css.

> Marp (also known as the Markdown Presentation Ecosystem) provides >>**an intuitive**<< experience for creating beautiful slide decks.

Most people will use some standart layouts and this breaks totaly the **intuitive** approch. I am very happy with the framework and html/css is a huge toolbox, but predetermined layouts should be a core thing. This might look like a minor thing but I realy believe this would open up for more people to use the framework. I cant go to a college and explain him "ye you can use markdown and make slides very easy buuuuuut you have to play around with very basic styling and waste your time on that"

**Reply by yhatt** [reply]
[Originally, Markdown does not have the concept of "columns".](https://stackoverflow.com/questions/30514408/have-two-columns-in-markdown/30520338#30520338:~:text=You%20can%27t%2C%20at,use%20HTML%20itself.) Is it really intuitive for those who write Markdown to provide some concept that does not exist in original Markdown?

The required slide design and layout is different depending on the presentation scene. So it's not good practice for Marp to arbitrarily define layouts like columns.

The theme system allows to predefine the design system of the slides according to the scene, including columns. The presentation author simply uses the theme as the theme author intended. There is no need to know the details of the style, and no time is wasted on it. :)

**Comment by theluke** [comment]
How do I change the number of columns for each slide instead of the beginning of the file?

**Reply by yhatt** [reply]
Following is another example of columns by using Flexbox instead of CSS Grid. The count of columns is changed dynamically by changing the number of `div` elements in the columns container.

```markdown
---
style: |
  .columns {
    display: flex;
    gap: 1rem;
  }
  .columns > div {
    flex: 1 1 0;
  }
---

# 2 Columns

<div class="columns">
<div>

## Column 1

Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nam placeat debitis labore quidem exercitationem architecto.

</div>
<div>

## Column 2

Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nam placeat debitis labore quidem exercitationem architecto.

</div>
</div>

---

# 3 Columns

<div class="columns">
<div>

## Column 1

Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nam placeat debitis labore quidem exercitationem architecto.

</div>
<div>

## Column 2

Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nam placeat debitis labore quidem exercitationem architecto.

</div>
<div>

## Column 3

Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nam placeat debitis labore quidem exercitationem architecto.

</div>
</div>
```

**Comment by vvmruder** [comment]
Thank you for this solution. It works well. I wonder if there would be an easy way to use this in combination with the code highlighting? So that on the left, there is some descriptive text and on the right some highlighted code snippet.
I tried some approaches but did not succeed. Especially when it is about highlighting XML code it ends up in an escape madness. May I ask for suggestions on that?

**Reply by yhatt** [reply]
@vvmruder

````markdown
---
marp: true
style: |
  .columns {
    display: grid;
    grid-template-columns: repeat(2, minmax(0, 1fr));
    gap: 1rem;

    /* Center align content vertically */
    align-items: center;
  }
---

# Multi columns in Marp slide

<div class="columns">
<div>

Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptas eveniet, corporis commodi vitae accusamus obcaecati dolor corrupti eaque id numquam officia velit sapiente incidunt dolores provident laboriosam praesentium nobis culpa.

</div>
<div>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Slide>
  <Heading>Multi columns</Heading>
  <Columns>
    <Column>Column 1</Column>
    <Column>Column 2</Column>
  </Columns>
</Slide>
```

</div>
</div>
````

![Example of columns with text and highlighted code block](https://github.com/user-attachments/assets/0e692e2c-ac0e-4d8e-ab63-00ac76fa43ff)

A blank line between opening `<div>` tag and the starting line of code block `` ```xml `` is always important to indicate the boundary of semantics, HTML and Markdown ([as an original solution does so](https://github.com/orgs/marp-team/discussions/192#discussioncomment-1516155)). If there was not, the code block would not render correctly and render as just a plain text.

> it provides a very simple and flexible way of including Markdown content inside HTML tags: simply separate the Markdown from the HTML using blank lines:
>
> _&mdash; [CommonMark Spec (4.6 HTML blocks)](https://spec.commonmark.org/0.31.2/#:~:text=it%20provides%20a%20very%20simple%20and%20flexible%20way%20of%20including%20Markdown%20content%20inside%20HTML%20tags%3A%20simply%20separate%20the%20Markdown%20from%20the%20HTML%20using%20blank%20lines%3A)_

**Reply by StructuralWizard** [reply]
Using Neobeam Theme here, I ended using a style override below to ensure the images have use the full width of the column. 

<style>
.container {
  display: grid;
  gap: 0.6rem;
  align-items: start;
}
.container.two {
  grid-template-columns: 1.2fr 1fr;
}
.container.three {
  grid-template-columns: 1.25fr 0.9fr 0.9fr;
}
.col ul {
  margin: 0;
  padding-left: 1.2rem;
}
.media img {
  width: 100%;
  max-width: 100% !important;
  max-height: 61vh;
  object-fit: contain;
  display: block;
  margin: 0 auto;
}
.single-media {
  width: 100%;
  max-height: 34vh;
  object-fit: contain;
  display: block;
  margin: 0.4rem auto 0;
}
</style>
